ntp-proto/src/packet.rs

```diff
@@ -275,11 +441,16 @@ impl NtpPacket {
         match version {
             4 => {
                 let (header, header_size) = NtpHeaderV4::deserialize(data)?;
-                if header_size != data.len() {
-                    debug!("Ignored potential extension fields or MAC");
-                }
+                let (efdata, fields_len) = ExtensionFieldData::deserialize(&data[header_size..])?;
+                let mac = if header_size + fields_len != data.len() {
+                    Some(Mac::deserialize(&data[header_size + fields_len..])?)
+                } else {
+                    None
+                };
                 Ok(NtpPacket {
                     header: NtpHeader::V4(header),
+                    efdata,
+                    mac,
                 })
             }
             _ => Err(PacketParsingError::InvalidVersion(version)),
```

Comment:
we also want to support version 3 now right?

Reply:
I will deal with that when rebasing, but in short, version 3 does not support extension fields, only a des-based mac.